// @storybook/addon-backgrounds/dist/BackgroundPanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.array.from");

require("core-js/modules/es6.regexp.to-string");

require("core-js/modules/es6.date.to-string");

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.array.is-array");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

require("core-js/modules/es6.array.map");

require("core-js/modules/es6.array.filter");

require("core-js/modules/es6.function.name");

require("core-js/modules/es6.array.find");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.keys");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.for-each");

require("core-js/modules/es6.string.trim");

var _global = require("global");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var _constants = _interopRequireDefault(require("./constants"));

var _Swatch = _interopRequireDefault(require("./Swatch"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Wrapper =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e137kbae0"
})({
  padding: 20
});
var Title =
/*#__PURE__*/
(0, _styled.default)("h5", {
  target: "e137kbae1"
})({
  fontSize: 16
});
var Pre =
/*#__PURE__*/
(0, _styled.default)("pre", {
  target: "e137kbae2"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    padding: '30px',
    display: 'block',
    background: theme.fillColor,
    marginTop: '15px',
    lineHeight: '1.75em'
  };
});
var List =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e137kbae3"
})({
  display: 'inline-block',
  padding: 15
});
var Item =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e137kbae4"
})({
  display: 'inline-block',
  padding: 5
});
var storybookIframe = 'storybook-preview-iframe';
var style = {
  iframe: {
    transition: 'background 0.25s ease-in-out'
  }
};
var defaultBackground = {
  name: 'default',
  value: 'transparent'
};
var instructionsHtml = "\nimport { storiesOf } from '@storybook/react';\nimport { withBackgrounds } from '@storybook/addon-backgrounds';\n\nstoriesOf('First Component', module)\n  .addDecorator(withBackgrounds([\n    { name: 'twitter', value: '#00aced' },\n    { name: 'facebook', value: '#3b5998\" },\n  ]))\n  .add(\"First Button\", () => <button>Click me</button>);\n".trim();

var Instructions = function Instructions() {
  return _react.default.createElement(Wrapper, null, _react.default.createElement(Title, null, "Setup Instructions"), _react.default.createElement("p", null, "Please add the background decorator definition to your story. The background decorate accepts an array of items, which should include a name for your color (preferably the css class name) and the corresponding color / image value."), _react.default.createElement("p", null, "Below is an example of how to add the background decorator to your story definition."), _react.default.createElement(Pre, null, _react.default.createElement("code", null, instructionsHtml)));
};

var BackgroundPanel =
/*#__PURE__*/
function (_Component) {
  _inherits(BackgroundPanel, _Component);

  function BackgroundPanel(props) {
    var _this;

    _classCallCheck(this, BackgroundPanel);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(BackgroundPanel).call(this, props));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "setBackgroundFromSwatch", function (background) {
      var api = _this.props.api;

      _this.updateIframe(background);

      api.setQueryParams({
        background: background
      });
    });

    _this.state = {
      backgrounds: []
    };
    return _this;
  }

  _createClass(BackgroundPanel, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      var _this$props = this.props,
          api = _this$props.api,
          channel = _this$props.channel;
      this.iframe = _global.document.getElementById(storybookIframe);

      if (!this.iframe) {
        throw new Error('Cannot find Storybook iframe');
      }

      Object.keys(style.iframe).forEach(function (prop) {
        _this2.iframe.style[prop] = style.iframe[prop];
      });
      channel.on(_constants.default.SET, function (data) {
        var backgrounds = _toConsumableArray(data);

        _this2.setState({
          backgrounds: backgrounds
        });

        var current = api.getQueryParam('background');
        var defaultOrFirst = backgrounds.find(function (x) {
          return x.default;
        }) || backgrounds[0]; // debugger;

        var foundBackground = current && backgrounds.find(function (bg) {
          return bg.name === decodeURI(current) || bg.value === current;
        });

        if (foundBackground) {
          _this2.updateIframe(foundBackground.value);
        } else if (defaultOrFirst) {
          _this2.updateIframe(defaultOrFirst.value);

          api.setQueryParams({
            background: defaultOrFirst.value
          });
        }
      });
      channel.on(_constants.default.UNSET, function () {
        _this2.setState({
          backgrounds: []
        });

        _this2.updateIframe('none');
      });
    }
  }, {
    key: "updateIframe",
    value: function updateIframe(background) {
      this.iframe.style.background = background;
    }
  }, {
    key: "render",
    value: function render() {
      var _this3 = this;

      var active = this.props.active;
      var _this$state$backgroun = this.state.backgrounds,
          backgrounds = _this$state$backgroun === void 0 ? [] : _this$state$backgroun;

      if (!active) {
        return null;
      }

      if (!backgrounds.length) return _react.default.createElement(Instructions, null);
      var hasDefault = backgrounds.filter(function (x) {
        return x.default;
      }).length;
      if (!hasDefault) backgrounds.push(defaultBackground);
      return _react.default.createElement(List, null, backgrounds.map(function (_ref2) {
        var value = _ref2.value,
            name = _ref2.name;
        return _react.default.createElement(Item, {
          key: "".concat(name, " ").concat(value)
        }, _react.default.createElement(_Swatch.default, {
          value: value,
          name: name,
          setBackground: _this3.setBackgroundFromSwatch
        }));
      }));
    }
  }]);

  return BackgroundPanel;
}(_react.Component);

exports.default = BackgroundPanel;
BackgroundPanel.propTypes = {
  active: _propTypes.default.bool.isRequired,
  api: _propTypes.default.shape({
    getQueryParam: _propTypes.default.func,
    setQueryParams: _propTypes.default.func
  }).isRequired,
  channel: _propTypes.default.shape({
    emit: _propTypes.default.func,
    on: _propTypes.default.func,
    removeListener: _propTypes.default.func
  })
};
BackgroundPanel.defaultProps = {
  channel: undefined
};
