// @storybook/addon-backgrounds/dist/Swatch.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.function.name");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button =
/*#__PURE__*/
(0, _styled.default)("button", {
  target: "e1qiy5ry0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    listStyle: 'none',
    backgroundColor: theme.barFill,
    textAlign: 'center',
    border: theme.mainBorder,
    borderRadius: theme.mainBorderRadius,
    color: 'inherit',
    cursor: 'pointer',
    display: 'inline-block',
    width: 175,
    verticalAlign: 'top',
    wordWrap: 'break-word',
    padding: 0,
    overflow: 'hidden'
  };
});
var Block =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e1qiy5ry1"
})(function (_ref2) {
  var bg = _ref2.bg,
      theme = _ref2.theme;
  return {
    height: 80,
    transition: 'opacity 0.25s ease-in-out',
    borderBottom: theme.mainBorder,
    background: bg,
    backgroundSize: 'cover',
    backgroundPosition: 'center'
  };
});
var Box =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e1qiy5ry2"
})({
  listStyle: 'none',
  paddingLeft: 10,
  paddingRight: 10
});
var Name =
/*#__PURE__*/
(0, _styled.default)("h4", {
  target: "e1qiy5ry3"
})({
  float: 'left',
  fontWeight: 'bold'
});
var Value =
/*#__PURE__*/
(0, _styled.default)("h4", {
  target: "e1qiy5ry4"
})({
  float: 'right',
  fontWeight: 'normal'
});

var Swatch = function Swatch(_ref3) {
  var name = _ref3.name,
      value = _ref3.value,
      setBackground = _ref3.setBackground;
  return _react.default.createElement(Button, {
    onClick: function onClick() {
      return setBackground(value);
    },
    onMouseDown: function onMouseDown(event) {
      return event.preventDefault();
    }
  }, _react.default.createElement(Block, {
    bg: value
  }), _react.default.createElement(Box, null, _react.default.createElement(Name, null, name, ":"), _react.default.createElement(Value, null, _react.default.createElement("em", null, value))));
};

Swatch.propTypes = {
  name: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired,
  setBackground: _propTypes.default.func.isRequired
};
var _default = Swatch;
exports.default = _default;
