// storybook-addon-i18next/manager.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var addons = _interopDefault(require('@storybook/addons'));
var PropTypes = _interopDefault(require('prop-types'));
var styled = _interopDefault(require('@emotion/styled'));
var components = require('@storybook/components');

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ("value" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function");
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      writable: true,
      configurable: true
    }
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) {
    return o.__proto__ || Object.getPrototypeOf(o);
  };
  return _getPrototypeOf(o);
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) {
    o.__proto__ = p;
    return o;
  };

  return _setPrototypeOf(o, p);
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return self;
}

function _possibleConstructorReturn(self, call) {
  if (call && (typeof call === "object" || typeof call === "function")) {
    return call;
  }

  return _assertThisInitialized(self);
}

function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest();
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArrayLimit(arr, i) {
  var _arr = [];
  var _n = true;
  var _d = false;
  var _e = undefined;

  try {
    for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance");
}

var ADDON_ID = 'i18next';
var PANEL_ID = "".concat(ADDON_ID, "/addon-panel");
var LANGUAGE_CHANGED_EVENT_ID = 'addon:i18next:languageChanged';
var CONFIGURE_EVENT_ID = 'addon:i18next:configure';

var Container = styled.div({
  padding: 15,
  width: '100%',
  boxSizing: 'border-box',
  height: '100%',
  overflow: 'auto'
});
Container.displayName = 'Container';

var Panel =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Panel, _React$Component);

  function Panel(props, context) {
    var _this;

    _classCallCheck(this, Panel);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Panel).call(this, props, context));
    _this.state = {
      language: null,
      languages: null
    };
    _this.configure = _this.configure.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.emitLanguageChanged = _this.emitLanguageChanged.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.handleLanguageSelect = _this.handleLanguageSelect.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    return _this;
  }

  _createClass(Panel, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var channel = this.props.channel;
      channel.on(CONFIGURE_EVENT_ID, this.configure);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      var channel = this.props.channel;
      channel.removeListener(CONFIGURE_EVENT_ID, this.configure);
    }
  }, {
    key: "configure",
    value: function configure(options) {
      var i18n = options.i18n,
          languages = options.languages;
      var language = i18n.language;
      this.setState({
        language: language,
        languages: languages
      });
    }
  }, {
    key: "emitLanguageChanged",
    value: function emitLanguageChanged() {
      var channel = this.props.channel;
      var language = this.state.language;
      channel.emit(LANGUAGE_CHANGED_EVENT_ID, language);
    }
  }, {
    key: "handleLanguageSelect",
    value: function handleLanguageSelect(event) {
      var _this2 = this;

      var language = event.target.value;
      var prevLang = this.state.language;

      if (prevLang !== language) {
        this.setState({
          language: language
        }, function () {
          _this2.emitLanguageChanged();
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var active = this.props.active;
      var _this$state = this.state,
          language = _this$state.language,
          languages = _this$state.languages;
      return active ? React.createElement(Container, null, React.createElement(components.Field, {
        label: "Language"
      }, React.createElement(components.Select, {
        size: "flex",
        value: language || '',
        onChange: this.handleLanguageSelect
      }, languages ? Object.entries(languages).map(function (_ref) {
        var _ref2 = _slicedToArray(_ref, 2),
            key = _ref2[0],
            name = _ref2[1];

        return React.createElement("option", {
          key: key,
          value: key
        }, name);
      }) : React.createElement("option", {
        value: "",
        disabled: true
      }, "(No Languages)")))) : null;
    }
  }]);

  return Panel;
}(React.Component);

Panel.propTypes = {
  active: PropTypes.bool.isRequired,
  channel: PropTypes.shape({
    on: PropTypes.func,
    emit: PropTypes.func,
    removeListener: PropTypes.func
  }).isRequired
};

/* eslint react/jsx-filename-extension:off */

var addChannel = function addChannel(api) {
  var channel = addons.getChannel();
  addons.addPanel(PANEL_ID, {
    title: 'i18next',
    // eslint-disable-next-line react/prop-types
    render: function render(_ref) {
      var active = _ref.active;
      return React.createElement(Panel, {
        active: active,
        channel: channel,
        api: api
      });
    }
  });
};

var init = function init() {
  addons.register(ADDON_ID, addChannel);
};

exports.init = init;
exports.addChannel = addChannel;
