// @storybook/addon-events/dist/components/Event.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

require("core-js/modules/es6.function.name");

require("core-js/modules/es6.string.trim");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactLifecyclesCompat = require("react-lifecycles-compat");

var _styled = _interopRequireDefault(require("@emotion/styled"));

var _formatJson = _interopRequireDefault(require("format-json"));

var _reactTextareaAutosize = _interopRequireDefault(require("react-textarea-autosize"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var StyledTextarea =
/*#__PURE__*/
(0, _styled.default)(_reactTextareaAutosize.default, {
  target: "e1qhyjxx0"
})({
  flex: '1 0 0',
  boxSizing: 'border-box',
  margin: '0 0 0 5px',
  verticalAlign: 'top',
  outline: 'none',
  border: '1px solid #c7c7c7',
  borderRadius: 2,
  fontSize: 13,
  padding: '8px 5px 7px 8px',
  color: 'rgb(51, 51, 51)',
  fontFamily: 'Arial, sans-serif',
  minHeight: '32px',
  resize: 'vertical'
}, function (_ref) {
  var shown = _ref.shown;
  return shown ? {} : {
    display: 'none'
  };
}, function (_ref2) {
  var failed = _ref2.failed;
  return failed ? {
    border: '1px solid #fadddd',
    backgroundColor: '#fff5f5'
  } : {};
});
var Button =
/*#__PURE__*/
(0, _styled.default)("button", {
  target: "e1qhyjxx1"
})({
  display: 'table-cell',
  textTransform: 'uppercase',
  letterSpacing: '3.5px',
  fontSize: 12,
  fontWeight: 'bolder',
  color: 'rgb(130, 130, 130)',
  border: '1px solid rgb(193, 193, 193)',
  textAlign: 'center',
  borderRadius: 2,
  padding: 5,
  cursor: 'pointer',
  paddingLeft: 8,
  margin: '0 0 0 5px',
  backgroundColor: 'inherit',
  verticalAlign: 'top',
  outline: 0
});
var Label =
/*#__PURE__*/
(0, _styled.default)("label", {
  target: "e1qhyjxx2"
})({
  display: 'table-cell',
  boxSizing: 'border-box',
  verticalAlign: 'top',
  paddingRight: 5,
  paddingTop: 7,
  textAlign: 'right',
  width: 100,
  fontWeight: '600'
});
var Wrapper =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "e1qhyjxx3"
})({
  display: 'flex',
  padding: 5,
  alignItems: 'flex-start',
  boxSizing: 'border-box',
  width: '100%'
});

function getJSONFromString(str) {
  try {
    return JSON.parse(str);
  } catch (e) {
    return str;
  }
}

var Item =
/*#__PURE__*/
function (_Component) {
  _inherits(Item, _Component);

  function Item() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Item);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Item)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      isTextAreaShowed: false
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "onChange", function (_ref3) {
      var value = _ref3.target.value;
      var newState = {
        payloadString: value
      };

      try {
        newState.payload = JSON.parse(value.trim());
        newState.failed = false;
      } catch (err) {
        newState.failed = true;
      }

      _this.setState(newState);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "onEmitClick", function () {
      var _this$props = _this.props,
          onEmit = _this$props.onEmit,
          name = _this$props.name;
      var payload = _this.state.payload;
      onEmit({
        name: name,
        payload: payload
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "onToggleEditClick", function () {
      _this.setState(function (_ref4) {
        var isTextAreaShowed = _ref4.isTextAreaShowed;
        return {
          isTextAreaShowed: !isTextAreaShowed
        };
      });
    });

    return _this;
  }

  _createClass(Item, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          title = _this$props2.title,
          name = _this$props2.name;
      var _this$state = this.state,
          failed = _this$state.failed,
          isTextAreaShowed = _this$state.isTextAreaShowed,
          payloadString = _this$state.payloadString;
      return _react.default.createElement(Wrapper, null, _react.default.createElement(Label, {
        htmlFor: "addon-event-".concat(name)
      }, title), _react.default.createElement(Button, {
        onClick: this.onEmitClick,
        disabled: failed,
        title: "Submit event"
      }, _react.default.createElement("span", {
        role: "img",
        "aria-label": "emit"
      }, "\uD83D\uDCE2")), _react.default.createElement(StyledTextarea, {
        shown: isTextAreaShowed,
        failed: failed,
        value: payloadString,
        onChange: this.onChange
      }), isTextAreaShowed ? _react.default.createElement(Button, {
        onClick: this.onToggleEditClick,
        title: "Close editing"
      }, _react.default.createElement("span", {
        role: "img",
        "aria-label": "close"
      }, "\u274C")) : _react.default.createElement(Button, {
        onClick: this.onToggleEditClick,
        title: "Edit event payload"
      }, _react.default.createElement("span", {
        role: "img",
        "aria-label": "edit"
      }, "\u270F\uFE0F")));
    }
  }]);

  return Item;
}(_react.Component);

_defineProperty(Item, "propTypes", {
  name: _propTypes.default.string.isRequired,
  title: _propTypes.default.string.isRequired,
  onEmit: _propTypes.default.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types, react/no-unused-prop-types
  payload: _propTypes.default.any
});

_defineProperty(Item, "defaultProps", {
  payload: {}
});

_defineProperty(Item, "getDerivedStateFromProps", function (_ref5, _ref6) {
  var payload = _ref5.payload;
  var prevPayload = _ref6.prevPayload;

  if (payload !== prevPayload) {
    var payloadString = _formatJson.default.plain(payload);

    return {
      failed: false,
      payload: getJSONFromString(payloadString),
      payloadString: payloadString,
      prevPayload: prevPayload
    };
  }

  return null;
});

(0, _reactLifecyclesCompat.polyfill)(Item);
var _default = Item;
exports.default = _default;
