// react-jsonschema-form-antd/lib/components/fields/TitleField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var REQUIRED_FIELD_SYMBOL = "*";

var titleStyle = {
  display: 'block',
  width: '100%',
  padding: '0',
  marginBottom: '20px',
  fontSize: '21px',
  lineHeight: 'inherit',
  color: '#333',
  border: '0',
  paddingBottom: '4px',
  borderBottom: '1px solid #e5e5e5'
};
function TitleField(props) {
  var id = props.id,
      title = props.title,
      required = props.required;

  var legend = required ? title + REQUIRED_FIELD_SYMBOL : title;
  return _react2.default.createElement(
    "div",
    { id: id, style: titleStyle },
    legend
  );
}

if (process.env.NODE_ENV !== "production") {
  TitleField.propTypes = {
    id: _propTypes2.default.string,
    title: _propTypes2.default.string,
    required: _propTypes2.default.bool
  };
}

exports.default = TitleField;
