// @storybook/addon-jest/dist/components/PanelTitle.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.array.reduce");

require("core-js/modules/es6.array.map");

require("core-js/modules/es6.array.filter");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var _provideJestResult = _interopRequireDefault(require("../hoc/provideJestResult"));

var _Indicator = _interopRequireDefault(require("./Indicator"));

var _colors = _interopRequireDefault(require("../colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrapper =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "eq6e62z0"
})({
  display: 'flex',
  alignItems: 'center'
});
var PanelName =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "eq6e62z1"
})({
  paddingLeft: 5
});

var PanelTitle = function PanelTitle(_ref) {
  var tests = _ref.tests;

  if (!tests) {
    return null;
  }

  var results = tests.map(function (report) {
    return report.result;
  }).filter(function (report) {
    return !!report;
  });
  var success = results.reduce(function (acc, result) {
    return acc && result.status === 'passed';
  }, true);
  var color = success ? _colors.default.success : _colors.default.error;
  return _react.default.createElement(Wrapper, null, _react.default.createElement(_Indicator.default, {
    color: results.length < tests.length ? _colors.default.warning : color,
    size: 10
  }), _react.default.createElement(PanelName, null, "Tests"));
};

PanelTitle.defaultProps = {
  tests: null
};
PanelTitle.propTypes = {
  tests: _propTypes.default.arrayOf(_propTypes.default.shape({
    result: _propTypes.default.object
  }))
};

var _default = (0, _provideJestResult.default)(PanelTitle);

exports.default = _default;
